### PR TITLE
[Storage] Fix#11043: Support remove whole container/share in az storage remove command

### LINKED ISSUE #11043: Error on batch-delete
## Describe the bug

**Command Name**
`az storage blob delete-batch`

**Errors:**
```
('Connection broken: OSError("(104, \'ECONNRESET\')",)', OSError("(104, 'ECONNRESET')",))
Traceback (most recent call last):
site-packages/urllib3/contrib/pyopenssl.py, ln 304, in recv_into
    return self.connection.recv_into(*args, **kwargs)
python3.6/site-packages/OpenSSL/SSL.py, ln 1822, in recv_into
    self._raise_ssl_error(self._ssl, result)
python3.6/site-packages/OpenSSL/SSL.py, ln 1622, in _raise_ssl_error
    raise WantReadError()
OpenSSL.SSL.WantReadError

...

storage/v2018_11_09/common/storageclient.py, ln 361, in _perform_request
    raise _wrap_exception(ex, AzureException)
azure.common.AzureException: ('Connection broken: OSError("(104, \'ECONNRESET\')",)', OSError("(104, 'ECONNRESET')",))
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az storage blob delete-batch -s {} --account-name {}`

## Environment Summary
```
Linux-4.15.0-66-generic-x86_64-with-debian-buster-sid
Python 3.6.5
Shell: bash

azure-cli 2.0.75

```

### PR DESCRIPTION
**Pull request by Juliehzl**
Fix #11043
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -61,6 +61,7 @@ Release History
 * GA Release Large File Shares property for storage account create and update command
 * GA Release User Delegation SAS token Support
 * Add new commands `az storage account blob-service-properties show` and `az storage account blob-service-properties update --enable-change-feed` to manage blob service properties for storage account.
+* Fix issue #11043: Support to remove container/share contents in `az storage remove` command
```
Comment (MyronFanQiu) [inline src/azure-cli/HISTORY.rst]: without --path/--blob?

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 2]
```diff
@@ -1083,6 +1083,8 @@ def validate_azcopy_remove_arguments(cmd, namespace):
     if valid_blob:
         client = blob_data_service_factory(cmd.cli_ctx, {
             'account_name': namespace.account_name})
+        if not blob:
+            blob = ''
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py]: server still need blob to be empty string. right? Is it also necessary for `path`?

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 3]
```diff
@@ -1072,8 +1072,8 @@ def validate_azcopy_remove_arguments(cmd, namespace):
     path = ns.pop('path', None)
 
     # ensure either a file or blob source is specified
-    valid_blob = container and blob and not share and not path
-    valid_file = share and path and not container and not blob
+    valid_blob = container and not share
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py]: Should we change these lines too?
```
usage_string = \
        'Invalid usage: {}. Supply only one of the following argument sets to specify source:' \	        'Invalid usage: {}. Supply only one of the following argument sets to specify source:' \
        '\n\t   --container-name  --name' \	        '\n\t   --container-name  --name' \
        '\n\tOR --share-name --path'	        '\n\tOR --share-name --path'
```

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py (hunk 2 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_validators.py]: For `path` part, it already support '' in CLI code. This is a legacy bug from previous member.

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py (hunk 3 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_validators.py]: Sure! Thanks for reminder!

File: src/azure-cli/HISTORY.rst (hunk 1 above)
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst]: remove whole container/share?

Comment (Juliehzl) [reply src/azure-cli/HISTORY.rst]: Sounds better!

Comment (Juliehzl) [reply src/azure-cli/HISTORY.rst]: Change to remove whole container/share. I think it is clear enough.

**Pull request merged**